tests/integration/upload_images.t

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/perl -w
+
+use ProductOpener::PerlStandards;
+
+use Test::More;
+use ProductOpener::APITest qw/:all/;
+use ProductOpener::Test qw/:all/;
+use ProductOpener::Products qw/:all/;
+use ProductOpener::TestDefaults qw/:all/;
+
+use File::Basename "dirname";
+
+use Storable qw(dclone);
+
+remove_all_users();
+
+remove_all_products();
+
+wait_application_ready();
+
+my $sample_products_images_path = dirname(__FILE__) . "/inputs/upload_images";
+
+# Create a normal user
+my $ua = new_client();
+
+my %create_user_args = (%default_user_form, (email => 'bob@gmail.com'));
+create_user($ua, \%create_user_args);
+
+# Create some products
+
+my @products = (
+
+	{
+		%{dclone(\%default_product_form)},
+		(
+			imgupload_front_en => ["$sample_products_images_path/front_en.3.full.jpg", 'front_en.3.full.jpg']
+		)
+	},
+
+);
+
+# create the products in the database
+foreach my $product_ref (@products) {
+	edit_product($ua, $product_ref);
+}
+
+my $tests_ref = [
+	{
+		test_case => 'post-product-image',
+		method => 'POST',
+		path => '/cgi/product_image_upload.pl',
+		form => {
+			code => "1234567890012",
+			imagefield => "front_en",
+			imgupload_front_en => ["$sample_products_images_path/1.jpg", '1.jpg'],
+		},
+		expected_status_code => 200,
+	},
+	{
+		test_case => 'get-product-image',
+		method => 'GET',
+		path => '/api/v2/product/1234567890012',
+		expected_status_code => 200,
+	},
+	{
+		test_case => 'post-image-too-small',
+		method => 'POST',
+		path => '/cgi/product_image_upload.pl',
+		form => {
+			code => "1234567890013",
+			imagefield => "front_en",
+			imgupload_front_en => ["$sample_products_images_path/small-img.jpg", 'small-img.jpg'],
+		}
+	},
+	{
+		test_case => 'get-image-too-small',
+		method => 'GET',
+		path => '/api/v2/product/1234567890013',
+
+	},
+	{
+		test_case => 'post-same-image-twice',
```

Comment:
for this product, there"s only 1 post request, so we need to duplicate the test case to upload the image twice